client/landing/stepper/declarative-flow/internals/steps-repository/playground/index.tsx

```diff
@@ -10,28 +10,90 @@ import { getBlueprintLabelForTracking } from './lib/blueprint';
 import type { Step as StepType } from '../../types';
 import './style.scss';
 
+const DEFAULT_PLAN_INTENT = 'plans-playground';
+
 export const PlaygroundStep: StepType = ( { navigation, flow } ) => {
 	const { submit } = navigation;
 	const playgroundClientRef = useRef< PlaygroundClient | null >( null );
 	const { __ } = useI18n();
 	const [ query ] = useSearchParams();
 
+	// For preventing double click on launch button
+	const [ isLaunching, setIsLaunching ] = useState( false );
+
+	// For calculating the intent, which would make launch wait for it as well
+	const [ pgIntent, setPgIntent ] = useState< string | null >( DEFAULT_PLAN_INTENT );
+	const [ calculatingIntent, setCalculatingIntent ] = useState( false );
+	const intentPromiseRef = useRef< Promise< void > | null >( null );
+
 	const setPlaygroundClient = ( client: PlaygroundClient ) => {
 		playgroundClientRef.current = client;
 	};
 
-	const launchSite = () => {
-		if ( ! submit ) {
+	const fetchIntent = () => {
+		const pgc = playgroundClientRef.current;
+		if ( ! pgc || calculatingIntent ) {
+			return;
+		}
+
+		setCalculatingIntent( true );
+		intentPromiseRef.current = pgc
+			.run( {
+				code: `<?php
+				require_once( '/wordpress/wp-load.php' );
+				die( 'plans-playground' ); // short-circuit the logic until we support free/premium plans
```

Comment:
@zaerl Short-circuting logic here

Reply:
A bit more explanation: This way our mode of calculation is tested and works, and its just logic is different.

Reply:
I think it's a good idea to create a folder to add the various PHP scripts, like `client/landing/.../playground/php-scripts/calculate-intent.ts`, that we will make to help in the onboarding phase. We will probably have more in the future, not just this one.

What do you think?

Reply:
Heya. Naive question - why are these php scripts needed? What would the alternative be?

Reply:
@chriskmnds 👋 We are running WordPress in the browser, so that's where this PHP script would execute, and not on WPCOM side.